fix(721): Remove hardcoded popover theme

bmansell opened the pull request:
- Remove all instances of hardcoded popover theme
- Remove prop `bg`/`background` from being passed into styles for `Arrow` and other inner components
- Substitute instances of `bg`/`background` with `color`
- Allow for specifying `borderColor` but will default to `border.base` if none is provided
- Added storybook example of theme colored Popovers
- Updated snapshots

#### Screenshots

##### New `Colors` storybook example
![Screen Shot 2020-05-06 at 11 28 01 AM](https://user-images.githubusercontent.com/12076625/81202797-b6020300-8f8c-11ea-9419-0644534f4274.png)

![Screen Shot 2020-05-06 at 11 28 10 AM](https://user-images.githubusercontent.com/12076625/81202799-b6020300-8f8c-11ea-80a6-d7a50fbde866.png)

![Screen Shot 2020-05-06 at 11 21 40 AM](https://user-images.githubusercontent.com/12076625/81202792-b4d0d600-8f8c-11ea-9a5c-42a5074d660c.png)

![Screen Shot 2020-05-06 at 11 21 51 AM](https://user-images.githubusercontent.com/12076625/81202796-b5696c80-8f8c-11ea-8946-48cc3920fcd2.png)

##### Playbook (pre-existing storybook example)
![Screen Shot 2020-05-06 at 11 18 48 AM](https://user-images.githubusercontent.com/12076625/81202669-8f43cc80-8f8c-11ea-97fa-854fc108c9f1.png)

Closes #721

pcln-james commented:
Based on the screenshots, the filled color popovers (success, alert, error) look a bit weird with the gray border, maybe we should make that the same color or just get rid of the border?

pcln-james commented on packages/popover/src/Arrow.js:
```diff
@@ -53,9 +52,10 @@ const ArrowAlignment = () =>
-  const bgColor = theme.colors[background]
-  const brColor = theme.colors[borderColor]
+const ArrowPlacement = ({ color, borderColor, ...props }) => {
+  let backgroundColorName = color.indexOf('.') !== -1 ? color : `${color}.base`
```
so this is to prevent somebody from doing `error.dark`?

pcln-james replied:
never mind, I got my ternary backwards 🤦

bmansell replied:
No, it's when something like `error` is passed as a color. The `getPaletteColor` usage in here isn't able to grab the corresponding background color from the palette to match the `Box` background color unless it's filled in manually here.

bmansell replied:
I can flip the condition around if it makes it easier to read though!

pcln-james commented on packages/popover/src/PopoverContent.js:
```diff
@@ -131,12 +140,11 @@ const PopperGuide = styled(Box)`
 `
 const ContentContainer = styled.section`
-  box-shadow: 0 0 0 1px ${({ theme, borderColor }) => theme.colors[borderColor]},
+  box-shadow: 0 0 0 1px ${props => getPaletteColor(props.borderColor)},
```
I think this needs to be `${props => getPaletteColor(props.borderColor)(props)}`

bmansell replied:
Fixed!

bmansell commented:
> Based on the screenshots, the filled color popovers (success, alert, error) look a bit weird with the gray border, maybe we should make that the same color or just get rid of the border?

The thing is, the border is created with the box shadow style and it looks a bit strange without that, so I'm not too sure what to change there.

achoboter commented on packages/popover/src/Arrow.js:
```diff
@@ -53,9 +52,10 @@ const ArrowAlignment = () =>
-  const bgColor = theme.colors[background]
-  const brColor = theme.colors[borderColor]
+const ArrowPlacement = ({ color, borderColor, ...props }) => {
+  let backgroundColorName = color.indexOf('.') !== -1 ? color : `${color}.base`
```
instead you can do this:
```
getPaletteColor(color, 'base')(props)
```
https://priceline.github.io/design-system/utilities/getPaletteColor#palette-color-driven-from-other-prop

bmansell replied:
Oh awesome, thank you!

bmansell replied:
Fixed!

pcln-james commented:
> @pcln-james
> 
> > Based on the screenshots, the filled color popovers (success, alert, error) look a bit weird with the gray border, maybe we should make that the same color or just get rid of the border?
> 
> The thing is, the border is created with the box shadow style and it looks a bit strange without that, so I'm not too sure what to change there.
> 
> _Edit_: Updated the storybook example to set the `borderColor` prop to the same as the `color` and it looks a lot better:
> <img alt="Screen Shot 2020-05-05 at 4 21 12 PM" width="615" src="https://user-images.githubusercontent.com/12076625/81117501-d1b5cc80-8eec-11ea-8bdf-ae3983bacd99.png">
> <img alt="Screen Shot 2020-05-05 at 4 21 23 PM" width="618" src="https://user-images.githubusercontent.com/12076625/81117502-d24e6300-8eec-11ea-808d-2bb0aa799f8b.png">
> <img alt="Screen Shot 2020-05-05 at 4 23 48 PM" width="635" src="https://user-images.githubusercontent.com/12076625/81117504-d24e6300-8eec-11ea-9797-a3d51877f32d.png">
> 
> Just not sure if having to set both is counter-intuitive to the purpose of `color` or not.
> 
> @sdalonzo what are your thoughts on this?

We could have the border color default to the value of the color prop if color is specified and border isn't?

bmansell commented:
> We could have the border color default to the value of the color prop if color is specified and border isn't?

@pcln-james Good point! I've tweaked the changes to the following:

- Remove the default props for `color` and `borderColor`
- Added a function to handle the logic for deciding which border color to use (`getBorderColorName`). 
- Changed the innermost `Box` component to be a styled component with a border radius value as the `ContainerComponent`'s border radius was getting obscured by the new color prop on `Box`. 

Screenshots have been updated in the original description! If you and @craigpalermo could take another peek that would be awesome, please and thanks!
